src/gsd-remote-display-manager.c

```diff
@@ -0,0 +1,295 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2012 Bastien Nocera <hadess@hadess.net>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "config.h"
+
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+
+#include <locale.h>
+
+#include <glib.h>
+#include <glib/gi18n.h>
+#include <gdk/gdk.h>
+#include <gdk/gdkx.h>
+#include <X11/Xatom.h>
+
+#include "gsd-remote-display-manager.h"
+
+enum
+{
+    PROP_0,
+    PROP_FORCE_DISABLE_ANIMATIONS
+};
+
+struct _GsdRemoteDisplayManager {
+        GObject parent;
+
+        /* Proxy for the force-disable-animations property */
+        gboolean      disabled;
+
+        GDBusProxy   *vino_proxy;
```

Comment:
I think we can drop the vino integration code. The plan is to drop vino support from GNOME (gnome-settings-daemon and Settings), and it's not in the build meta any longer.

Reply:
Where's the gnome-remote-desktop code to replace it?

Reply:
I'll wait for those changes to happen in g-s-d, and copy a newer version at some point.

Reply:
I don't think that file will remain at all in g-s-d. GNOME Shell will learn how to decide when to disable animations; g-s-d will only relay this information via Xsettings, and it I assume it'll just be absorbed into some other file. As in, there will be no "remote display manager" left in g-s-d in 3.36.

Reply:
How will the information be communicated from gnome-shell to gsd?

Reply:
gnome-remote-desktop will set a property on a remote desktop session. This will be communicated via MetaRemoteAccessControl (currently used for the remote desktop system menu section). The shell would then decide whether animations should be disabled or not, and expose that via D-Bus. Both gsd and x-d-p-gtk would then get the value from there.

Reply:
So there will be new gnome-remote-desktop code to replace the existing code in g-s-d. I'm not too sure why you'd say "I don't think that file will remain at all in g-s-d." when it will need to be adapted to use the property exported by gnome-remote-desktop instead of the one exported by vino...

Reply:
Except that it won't have anything to do with "remote displays", it'll just forward a boolean it only knows the name of.

Reply:
Hardly a big problem in this case.

Reply:
ok, sounds like we'll handle this for 3.36